src/core/vdom/helpers/resolve-async-component.js

```diff
@@ -61,9 +61,21 @@ export function resolveAsyncComponent (
     const contexts = factory.contexts = [context]
     let sync = true
 
-    const forceRender = () => {
+    const forceRender = (renderCompleted: boolean) => {
       for (let i = 0, l = contexts.length; i < l; i++) {
         contexts[i].$forceUpdate()
+
+        if (!renderCompleted) {
+          continue
+        }
+
+        const currContext = contexts[i]
+        const contextIdx = contexts.indexOf(currContext)
```

Comment:
isn't `contextInx === i`?

Reply:
@wa3l - good catch. The correct value of `i` is preserved when used in `$nextTick` if I just assign it to `contextIdx`.
I have updated the diff to include this refactor